packages/swap/src/utils/getPoolQuote.ts

```diff
@@ -0,0 +1,81 @@
+import { bigintMin } from '@/shared/functions';
+import { QuoteQueryResponse, SwapFee } from '@/shared/schemas';
+import {
+  buildFee,
+  calculateIncludedSwapFees,
+  estimateIngressEgressFeeAssetAmount,
+} from '@/swap/utils/fees';
+import { estimateSwapDuration } from '@/swap/utils/swap';
+import logger from './logger';
+import { getMinimumEgressAmount, getEgressFee } from './rpc';
+import ServiceError from './ServiceError';
+import { getSwapRate } from './statechain';
+import { InternalAsset } from '../client';
+import { checkPriceWarning } from '../pricing/checkPriceWarning';
+import { QuoteType } from '../quoting/Quoter';
+
+export default async function getPoolQuote(
+  srcAsset: InternalAsset,
+  destAsset: InternalAsset,
+  swapInputAmount: bigint,
+  ingressEgressFeeIsGasAssetAmount: boolean,
+  fees: SwapFee[],
+): Promise<QuoteQueryResponse & { quoteType: QuoteType }> {
+  const includedFees: SwapFee[] = [...fees];
+
+  const start = performance.now();
+
+  const quote = await getSwapRate({ srcAsset, destAsset, amount: swapInputAmount });
+
+  const lowLiquidityWarning = await checkPriceWarning({
+    srcAsset,
+    destAsset,
+    srcAmount: swapInputAmount,
+    destAmount: BigInt(quote.outputAmount),
+  });
+
+  const quoteSwapFees = await calculateIncludedSwapFees(
+    srcAsset,
+    destAsset,
+    swapInputAmount,
+    quote.intermediateAmount,
+    quote.outputAmount,
+  );
+  includedFees.push(...quoteSwapFees);
+
+  let egressFee = await getEgressFee(destAsset);
```

Comment:
feels a bit messy to calculate these things in two different places but lets just hope taht we dont have this file for long anyway 😅

Reply:
Should be easy enough to merge, can try it in a follow up